### PR TITLE
Method to clear the full cache

### PR DESCRIPTION
**Pull request by rizwankce**
Check out the original blog [post](http://khanlou.com/2017/10/the-flat-cache/) which gave us the idea of `FlatCache`. The author have a Clear cache method just to clear out the whole cache.

### CODE REVIEW

File: FlatCache/FlatCache.swift [hunk 1]
```diff
@@ -136,5 +136,9 @@ public final class FlatCache {
         return storage[key] as? T
     }
 
+    public func clearCache() {
```
Comment (rizwankce) [inline FlatCache/FlatCache.swift]: Kind of feel the name is weird from the caller site. For now it looks like 
```swift
cache.clearCache()
```
May be we could short down to 
```swift
cache.clear()
```

File: FlatCache/FlatCache.swift [hunk 2]
```diff
@@ -136,5 +136,9 @@ public final class FlatCache {
...
+        storage = [:]
+    }
```
Comment (rizwankce) [inline FlatCache/FlatCache.swift]: Also, Do we need to let the `listeners` know that cache is being cleared. Im not sure here

File: FlatCache/FlatCache.swift (hunk 1 above)
Comment (toohotz) [reply FlatCache/FlatCache.swift]: The listeners should be told that the cache that they're observing is being cleared (in my opinion at least).  In my PR #4 I do go through the listeners and notify the ones that are listening that an item is being removed from the cache.  I would say that removing a single item logic should come before the clearing the cache as that is essentially performing multiple "_removes_" which would update the listeners as well.  @rnystrom thoughts?

**Comment by rizwankce** [comment]
Updated to let all the listeners know that cache is being cleared and renamed `clearCache` to `clear`

**Comment by rnystrom** [comment]
Can you add the main thread assert? This cache isn’t thread safe yet so it needs some warnings.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rizwankce** [comment]
@rnystrom Yup! Added the main thread assert 👍

**Pull request merged**